src/odb/src/db/dbModule.cpp

```diff
@@ -543,6 +543,372 @@ dbBlock* dbModule::getOwner()
   return (dbBlock*) obj->getOwner();
 }
 
+// Create a "deep" and unique copy of old_module based on
+// its instance context.   All ports, instances, mod nets and
+// parent/child IO will be copied.  Connections that span
+// multiple modules needs to be done outside this API.
+dbModule* dbModule::copy(dbModule* old_module, dbModInst* new_mod_inst)
+{
+  dbBlock* block = old_module->getOwner();
+  // Uniquify new module name as "module_name.inst_name"
+  std::string orig_cell_name(old_module->getName());
+  size_t dot_index = orig_cell_name.find_first_of('.');
+  if (dot_index != std::string::npos) {
+    orig_cell_name = orig_cell_name.substr(0, dot_index);
+  }
+  std::string new_cell_name
+      = orig_cell_name + '.' + std::string(new_mod_inst->getName());
```

Comment:
I respectfully suggest factoring out the uniquification from the copy and also not passing in the instance.
So have two functions:
1. Get the dbModule "shell" --new_db_module = makeUniqueDbModule(orig_db_module -> name) (this builds the next uniquified named dbModule with nothing in it). I respectfully suggest using the same code as before. Also don't use "." in verilog names ("." has meaning).
2. Then deep copy  deepCopyContents(dbModule* original , dbModule* new). (this fills out the new dbModule by deep copying contents from original).